sdk/src/keystore.rs

```diff
@@ -0,0 +1,89 @@
+use crate::errors::KeystoreError;
+use dirs::data_dir;
+use eth_keystore::{decrypt_key, new as eth_keystore_new};
+use lazy_static::lazy_static;
+use rand::rngs::OsRng;
+use secp256k1::SecretKey;
+use std::fs;
+use std::path::Path;
+
+lazy_static! {
+    pub static ref KEYSTORE_DEFAULT_PATH: String = {
+        let data_dir = data_dir().expect("Failed to get base directories");
+        data_dir
+            .join("rex/keystores/")
+            .to_str()
+            .expect("Failed to convert path to string")
+            .to_owned()
+    };
+}
+
+/// Creates a new keystore in the given path and name using the password.
+/// If no path is provided, uses KEYSTORE_DEFAULT_PATH.
+/// If no name is provided, generates a random one.
+/// Returns the SecretKey generated a the UUID of the filesystem.
+pub fn create_new_keystore<S>(
+    path: Option<&str>,
+    name: Option<&str>,
+    password: S,
+) -> Result<(SecretKey, String), KeystoreError>
+where
+    S: AsRef<[u8]>,
+{
+    let path = match path {
+        Some(p) => Path::new(p),
+        None => {
+            // check if default path exists or create it
+            let p = Path::new(KEYSTORE_DEFAULT_PATH.as_str());
+            if !p.exists() {
+                fs::create_dir_all(KEYSTORE_DEFAULT_PATH.as_str())
+                    .map_err(|e| KeystoreError::ErrorCreatingDefaultDir(e.to_string()))?;
+            }
+            p
+        }
+    };
```

Comment:
This could be written simpler

```suggestion
    let path = path.map_or(Path::new(KEYSTORE_DEFAULT_PATH.as_str()), Path::new);

    if !path.exists() {
        fs::create_dir_all(KEYSTORE_DEFAULT_PATH.as_str())
            .map_err(|e| KeystoreError::ErrorCreatingDefaultDir(e.to_string()))?;
    }
```

Reply:
With that logic we are checking if the path exists (default or not) and creating a default path

Reply:
You're right, we should create `path` instead.

Reply:
Just updated the suggestion.

I forgot to mention that we weren't creating the provided path if it didn't exist.

Reply:
Done!